lib/ldap.js

```diff
@@ -0,0 +1,65 @@
+const ldap = require('ldapjs')
+const client = ldap.createClient({
+  url: process.env.LDAP_URL
+})
+
+function connect () {
+  return new Promise((resolve, reject) => {
+    client.bind(process.env.LDAP_USERNAME, process.env.LDAP_PASSWORD, (err) => {
+      if (err) {
+        return reject(err)
+      }
+
+      resolve()
+    })
+  })
+}
+
+function search (filter) {
+  return new Promise((resolve, reject) => {
+    const options = {
+      filter,
+      scope: 'sub',
+      paged: true,
+      sizeLimit: 1000,
+      attributes: []
```

Comment:
One could use `attributes` to only fetch "required" data e.g. `attributes: ['ugLadok3StudentUid']`

Reply:
Good point!